testing/citest/tests/openstack_smoke_test.py

```diff
@@ -0,0 +1,258 @@
+        self.new_post_operation(
+            title='create_security_group', data=payload,
+            path='applications/{app}/tasks'.format(app=self.TEST_APP)),
+        contract=builder.build())
+
+  def delete_a_security_group(self):
+    """Creates OsContract for deleteServerGroup.
+
+    To verify the operation, we just check that the spinnaker security group
+    for the given application was deleted.
+    """
+    #Get ID of the created security group
+    os_agent = os.OsAgent(None)
+    data = os_agent.get_resource('security group', self.TEST_SECURITY_GROUP)
+    security_group_id = data['id']
+
+    payload = self.agent.make_json_payload_from_kwargs(
+        job=[{
+            "Provider": "openstack",
+            "id": security_group_id,
+            "region": self.bindings['OS_REGION_NAME'],
+            "regions": [self.bindings['OS_REGION_NAME']],
+            "account": self.bindings['SPINNAKER_OS_ACCOUNT'],
+            "securityGroupName": self.TEST_SECURITY_GROUP,
+            "cloudProvider": "openstack",
+            "type": "deleteSecurityGroup",
+            "user": self.bindings['TEST_OS_USERNAME']
+        }],
+        application=self.TEST_APP,
+        description='Delete Security Group: : ' + self.TEST_SECURITY_GROUP)
+
+    builder = os.OsContractBuilder(self.os_observer)
+    (builder.new_clause_builder(
+        'Security Group Deleted', retryable_for_secs=30)
+     .show_resource('security group', self.TEST_SECURITY_GROUP,
+                    no_resources_ok=True)
+     .excludes_path_eq('name', self.TEST_SECURITY_GROUP)
+    )
+    return st.OperationContract(
+        self.new_post_operation(
+            title='delete_security_group', data=payload,
+            path='applications/{app}/tasks'.format(app=self.TEST_APP)),
+        contract=builder.build())
+
+class OpenStackSmokeTest(st.AgentTestCase):
+  """The test fixture for the OpenStackSmokeTest.
+
+  This is implemented using citest OperationContract instances that are
+  created by the OpenStackSmokeTestScenario.
+  """
+  # pylint: disable=missing-docstring
+  @property
+  def scenario(self):
+    return citest.base.TestRunner.global_runner().get_shared_data(
+        OpenStackSmokeTestScenario)
+
+  def test_a_create_app(self):
+    self.run_test_case(self.scenario.create_app())
+
+
+  def test_b_create_security_group(self):
+    self.run_test_case(self.scenario.create_a_security_group())
+
+
+  def test_z_delete_app(self):
+    self.run_test_case(self.scenario.delete_app(),
+                       retry_interval_secs=8, max_retries=8)
+
+  def test_y_delete_security_group(self):
+    self.run_test_case(self.scenario.delete_a_security_group(),
+                       retry_interval_secs=8, max_retries=8)
+
+def main():
+  """Implements the main method running this smoke test."""
+
+  defaults = {
+      'TEST_STACK': OpenStackSmokeTestScenario.DEFAULT_TEST_ID,
+      'TEST_APP': 'openstacksmoke' + OpenStackSmokeTestScenario.DEFAULT_TEST_ID
```

Comment:
I've found it useful to have test or tst in the name so it is easier to recognize and clean things up should tests fail and leave things behind (or have bugs where they create but dont reap).

So you might want to prefix this as 'os-smoketest' or 'openstack-smoketest'